client/js/widget-manager.js

```diff
@@ -10,7 +10,7 @@
 var $ = require('jquery');
 var Widgets = require('jupyter-js-widgets');
 
-    var WidgetManager = function(kernel, msgHandler) {
+    var WidgetManager = function(kernel, kernelname, msgHandler) {
```

Comment:
`IKernel` (type of `kernel`) has a `name` property, but it always seems to come back as empty string in my testing.